frontend/src/components/Chatbot.tsx

```diff
@@ -14,45 +16,159 @@ import { v4 as uuidv4 } from 'uuid';
 export const Chatbot = () => {
   const [isOpen, setIsOpen] = useState(false);
   const [input, setInput] = useState('');
+  const [sessions, setSessions] = useState<ChatSession[]>([]);
+  const [currentSessionId, setCurrentSessionId] = useState<string | null>(null);
+  const [isSidebarOpen, setIsSidebarOpen] = useState<boolean>(false);
 
   /**
    * Messages shown in the chat.
    * Initialized from sessionStorage to keep messages between refreshes.
    */
-  const [messages, setMessages] = useState<Message[]>(() => {
-    const saved = sessionStorage.getItem('chatbot-messages');
-    return saved ? JSON.parse(saved) : [];
-  });
   const [loading, setLoading] = useState(false);
 
+  /**
+   * Saving the chat sessions in the session storage only 
+   * when the component umounts to avoid continuos savings.
+   */
+  useEffect(() => {
+    return () => {
+      sessionStorage.setItem('chatbot-sessions', JSON.stringify(sessions));
+    };
+  }, [sessions]);
+
+  /**
+   * Loading the chat sessions from the session storage at mount time.
+   */
   useEffect(() => {
-    sessionStorage.setItem('chatbot-messages', JSON.stringify(messages));
-  }, [messages]);
+    const saved = sessionStorage.getItem('chatbot-sessions');
+    if (saved && saved.length > 0) {
+      try {
+        const parsed = JSON.parse(saved);
+        setSessions(parsed);
+        if (parsed.length > 0) {
+          setCurrentSessionId(parsed[0].id);
```

Comment:
I'm thinking we should store and use the last visited session, instead of the first session. What do you think?